lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs

```diff
@@ -127,7 +127,66 @@ public void TestDeleteDirectoryRecursive()
         public void TestCreateUriForPath()
         {
             var testContext = new TestContext();
-            Assert.Equal(FakeUri, testContext.GetAzureFileSystem().CreateUriForPath(FakeUri.LocalPath));
+            var tests = new[]
+            {
+                new
+                {
+                    Path = FakeUri.AbsoluteUri,
+                    Uri = FakeUri,
+                    ExpectedExceptionType = (Type)null
+                },
+                new
+                {
+                    Path = "http://storageacct.blob.core.windows.net/container/folder1/folder2/file.txt",
+                    Uri = new Uri("http://storageacct.blob.core.windows.net/container/folder1/folder2/file.txt"),
+                    ExpectedExceptionType = (Type)null
+                },
+                new
+                {
+                    Path = "container/folder1/folder2/file.txt",
+                    Uri = new Uri("http://storageacct.blob.core.windows.net/container/folder1/folder2/file.txt"),
+                    ExpectedExceptionType = (Type)null
+                },
+                new
+                {
+                    // Container name must be atleast 3 characters
+                    Path = "c1/folder1/folder2/file.txt",
+                    Uri = new Uri("http://storageacct.blob.core.windows.net/container/folder1/folder2/file.txt"),
+                    ExpectedExceptionType = typeof(ArgumentException)
+                },
+                new
+                {
+                    // Container name cannot contain a colon
+                    Path = "c:/folder1/folder2/file.txt",
+                    Uri = (Uri)null,
+                    ExpectedExceptionType = typeof(ArgumentException)
+                }
+            };
+
+            foreach (var test in tests)
+            {
+                if (test.ExpectedExceptionType == null)
+                {
+                    Exception actualException = null;
+                    Uri resultUri = null;
+
+                    try
+                    {
+                        resultUri = testContext.GetAzureFileSystem().CreateUriForPath(test.Path);
+                    }
+                    catch (Exception ex)
+                    {
+                        actualException = ex;
+                    }
+
+                    Assert.Null(actualException);
```

Comment:
Shouldn't be testing the exception for null. Instead put an Assert.Fail inside the exception itself.

Reply:
This code was removed when I refactored the tests into individual tests. Hence no longer an issue.